Apps/WebClient/src/Server/Listeners/BannerListener.cs

```diff
@@ -0,0 +1,142 @@
+    using System.Threading.Tasks;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Models;
+    using HealthGateway.WebClient.Models;
+    using HealthGateway.WebClient.Services;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Logging;
+    using Npgsql;
+
+    /// <summary>
+    /// Implements the abstract DB Listener and listens on the BannerChange channel.
+    /// </summary>
+    public class BannerListener : BackgroundService
+    {
+        private const string DeleteAction = "DELETE";
+        private const string UpdateAction = "UPDATE";
+        private const string InsertAction = "INSERT";
+        private const string Channel = "BannerChange";
+
+        private readonly ILogger<BannerListener> logger;
+        private readonly IConfiguration configuration;
+        private readonly IServiceProvider services;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BannerListener"/> class.
+        /// </summary>
+        /// <param name="logger">The injected logger.</param>
+        /// <param name="configuration">The injected configuration.</param>
+        /// <param name="services">The set of application services available.</param>
+        public BannerListener(
+            ILogger<BannerListener> logger,
+            IConfiguration configuration,
+            IServiceProvider services)
+        {
+            this.logger = logger;
+            this.configuration = configuration;
+            this.services = services;
+        }
+
+        private ICommunicationService? CommunicationService { get; set; }
+
+        /// <summary>
+        /// Creates a new DB Connection for push notifications from the DB for a specific channel.
+        /// </summary>
+        /// <param name="stoppingToken">The cancellation token to use.</param>
+        /// <returns>The task.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "Abstract class property")]
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            this.logger.LogDebug($"Registering Channel Notification on channel {Channel}");
+            stoppingToken.Register(() =>
+                this.logger.LogInformation($"DBChangeListener exiting..."));
+            try
+            {
+                using var scope = this.services.CreateScope();
+                GatewayDbContext dbContext = scope.ServiceProvider.GetRequiredService<GatewayDbContext>();
+                this.CommunicationService = scope.ServiceProvider.GetRequiredService<ICommunicationService>();
+                NpgsqlConnection con = (NpgsqlConnection)dbContext.Database.GetDbConnection();
```

Comment:
Do you think using "using" block is better to protect our db connection from leaking?